entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -1127,3 +1127,468 @@ In addition to the enhancements described above, EF8 Preview 2 also includes som
+
+The first option has advantages in many situations--we'll take a quick look at it at the end of this section. However, it's not a natural representation of the data in the model, and if what you really have is a collection of a primitive type, then the second option can be more effective.
+
+Starting with Preview 4, EF8 now includes built-in support for the second option, using JSON as the serialization format. JSON works well for this since modern relational databases include built-in mechanisms for querying and manipulating JSON, such that the JSON column can, effectively, be treated as a table when needed, without the overhead of actually creating that table. These same mechanisms allow JSON to be passed in parameters and then used in similar way to table-valued parameters in queries--more about this later.
+
+> [!TIP]
+> The code shown here comes from [PrimitiveCollectionsSample.cs](https://github.com/dotnet/EntityFramework.Docs/tree/main/samples/core/Miscellaneous/NewInEFCore8/PrimitiveCollectionsSample.cs).
+
+#### Primitive collection properties
+
+EF Core can map any `IEnumerable<T>` property, where `T` is a primitive type, to a JSON column in the database. This is done by convention for public properties which have both a getter and a setter. For example, all properties  in the following entity type are mapped to JSON columns by convention:
+
+```csharp
+public class PrimitiveCollections
+{
+    public IEnumerable<int> Ints { get; set; }
+    public ICollection<string> Strings { get; set; }
+    public ISet<DateTime> DateTimes { get; set; }
+    public IList<DateOnly> Dates { get; set; }
+    public uint[] UnsignedInts { get; set; }
+    public List<bool> Booleans { get; set; }
+    public List<Uri> Urls { get; set; }
+}
+```
+
+> [!NOTE]
+> What do we mean by "primitive type" in this context? Essentially, something that the database provider knows how to map, using some kind of value conversion if necessary. For example, in the entity type above, the types `int`, `string`, `DateTime`, `DateOnly` and `bool` are all handled without conversion by the database provider. SQL Server does not have native support for unsigned ints or URIs, but `uint` and `Uri` are still treated as primitive types because there are [built-in value converters](xref:core/modeling/value-conversions#built-in-converters) for these types.
+
+By default, EF Core uses an unconstrained Unicode string column type to hold the JSON, since this protects against data loss with large collections. However, on some database systems, such as SQL Server, specifying a maximum length for the string can improve performance. This, along with other column configuration, can be done [in the normal way](xref:core/modeling/entity-properties). For example:
+
+```csharp
+modelBuilder
+    .Entity<PrimitiveCollections>()
+    .Property(e => e.Booleans)
+    .HasMaxLength(1024)
+    .IsUnicode(false);
```

Comment:
Maybe consider not showing IsUnicode(false).. I get that we want to show facet config in general, but some people may copy paste the whole thing or think that it's somehow needed for max length

Reply:
Doesn't it make sense to not use Unicode for a JSON list of bool values?

Reply:
For bools (non-strings in general) yeah, you're right - it's just that before you discuss only the max length part. Ok with it as-is as well.